source/base/data_out_base.cc

```diff
@@ -5571,11 +5593,14 @@ namespace DataOutBase
         const auto first_component = std::get<0>(range);
         const auto last_component  = std::get<1>(range);
         const auto name            = std::get<2>(range);
+        const bool is_tensor =
+          (std::get<3>(range) ==
+           DataComponentInterpretation::component_is_part_of_tensor);
```

Comment:
I guess there should be an additional assertion here, that if `!is_tensor` then `std::get<3>(range) == DataComponentInterpretation::component_is_part_of_vector`

Reply:
>  there should be an additional assertion here

why? `is_tensor` just a short-cut for range being a tensor. Eventually somebody will add another interpretation so I don't see the reason of having this assert.